uber/site_sections/preregistration.py

```diff
@@ -109,6 +109,14 @@ def form(self, session, message='', edit_id=None, **params):
             params.setdefault('badges', group.badges)
         else:
             attendee = session.attendee(params, ignore_csrf=True, restricted=True)
+            if attendee.found_how:
+                code = session.query(PromoCode).filter(PromoCode.code == attendee.found_how).first()
+                attendee.promo_code = code
+                attendee.promo_code_id = code.id
```

Comment:
I don't think you need to set both of these here. Just `attendee.promo_code = code` should work.